### PR TITLE
refactor(theme-toggle): simplify transitions and unify naming

### PR DESCRIPTION
**Pull request by its-monotype**
## Before

https://user-images.githubusercontent.com/79363260/226901630-402ef650-d40d-43ae-9e80-693fa3002b4c.mp4

## After

https://user-images.githubusercontent.com/79363260/226901676-8da018f3-29e0-43aa-999a-a0acc58d45d5.mp4

### CODE REVIEW

File: templates/next-template/components/theme-toggle.tsx [hunk 1]
```diff
@@ -17,12 +17,12 @@ export function ThemeToggle() {
...
         <Button variant="ghost" size="sm">
-          <Icons.sun className="rotate-0 scale-100 transition-all hover:text-slate-900 dark:-rotate-90 dark:scale-0 dark:text-slate-400 dark:hover:text-slate-100" />
-          <Icons.moon className="absolute rotate-90 scale-0 transition-all hover:text-slate-900 dark:rotate-0 dark:scale-100 dark:text-slate-400 dark:hover:text-slate-100" />
+          <Icons.sun className="rotate-0 scale-100 transition-transform dark:-rotate-90 dark:scale-0" />
+          <Icons.moon className="absolute rotate-90 scale-0 transition-transform dark:rotate-0 dark:scale-100" />
           <span className="sr-only">Toggle theme</span>
         </Button>
       </DropdownMenuTrigger>
-      <DropdownMenuContent align="end" forceMount>
+      <DropdownMenuContent align="end">
```
Comment (shadcn) [inline templates/next-template/components/theme-toggle.tsx:25]: Ah thanks for catching this.

File: apps/www/components/site-header.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { CommandMenu } from "@/components/command-menu"
 import { Icons } from "@/components/icons"
 import { MainNav } from "@/components/main-nav"
 import { MobileNav } from "@/components/mobile-nav"
-import { ModeToggle } from "@/components/mode-toggle"
+import { ThemeToggle } from "@/components/theme-toggle"
```
Comment (shadcn) [inline apps/www/components/site-header.tsx:8]: What are your thoughts on keeping the name to **Mode**Toggle? Here's how I think of it (given that I'll be adding CSS variables soon). 

* Mode is for light and dark mode.
* Theme is for visual like colors, typography...etc.
* A theme can exist in light or dark mode.

Comment (its-monotype) [reply apps/www/components/site-header.tsx:8]: Sounds good but the name "ModeToggle" may not be clear enough to indicate its purpose. To make it more descriptive, consider renaming it as "ColorModeToggle." or "LightDarkToggle" Maybe you plan to add more predefined themes, such as coffee light, coffee dark, midnight light, and midnight dark so it's should be also considered when naming, and in that case you can keep the name "ThemeToggle" for changing the theme. For customizing colors, consider using a name like "ColorCustomizer" or or "ColorsSettings". But if you only planning dark and light, then "mode" would be fine for it, maybe just rename it to "ColorModeToggle" or smith like this.

**Comment by its-monotype** [comment]
Closed by #180

**Pull request closed without merging**